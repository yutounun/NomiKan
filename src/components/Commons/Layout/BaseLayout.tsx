import React from "react";

class Props {
  children?: React.ReactNode = null;
}

function BaseLayout({ children }: Props) {
  return <main data-testid="main">{children}</main>;
}

export default BaseLayout;
