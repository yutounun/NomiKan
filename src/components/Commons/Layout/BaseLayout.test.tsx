import { render, screen } from "@testing-library/react";
import React from "react";
import BaseLayout from "./BaseLayout";

describe("BaseLayout", () => {
  it("renders correctly", () => {
    render(<BaseLayout>This is content</BaseLayout>);
    expect(screen.getByTestId("main")).toBeInTheDocument();
  });

  test("renders the component with the correct prop", () => {
    const textProp = "This is content";
    render(<BaseLayout>{textProp}</BaseLayout>);
    const textElement = screen.getByText(textProp);
    expect(textElement).toBeInTheDocument();
  });

  // it("takes a snapshot", () => {
  //   const { asFragment } = render(<BaseLayout />);
  //   expect(asFragment()).toMatchSnapshot();
  // });
});
