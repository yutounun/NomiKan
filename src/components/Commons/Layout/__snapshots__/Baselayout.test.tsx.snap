// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseLayout takes a snapshot 1`] = `
<DocumentFragment>
  <main
    data-testid="main"
  />
</DocumentFragment>
`;
